motionplan/tpspace/ptgGroupFrame.go

```diff
@@ -151,18 +152,26 @@ func (pf *ptgGroupFrame) MarshalJSON() ([]byte, error) {
 	return nil, nil
 }
 
-// Inputs are: [0] index of PTG to use, [1] index of the trajectory within that PTG, and [2] distance to travel along that trajectory.
+// Inputs are: [0] index of PTG to use, [1] index of the trajectory within that PTG, [2] starting point on the trajectory, and [3] distance
+// to travel along that trajectory.
 func (pf *ptgGroupFrame) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
 	if len(inputs) != len(pf.DoF()) {
 		return nil, referenceframe.NewIncorrectInputLengthError(len(inputs), len(pf.DoF()))
 	}
 
 	ptgIdx := int(math.Round(inputs[ptgIndex].Value))
 
-	pose, err := pf.solvers[ptgIdx].Transform([]referenceframe.Input{inputs[trajectoryAlphaWithinPTG], inputs[distanceAlongTrajectoryIndex]})
+	pose, err := pf.solvers[ptgIdx].Transform([]referenceframe.Input{inputs[trajectoryAlphaWithinPTG], inputs[endDistanceAlongTrajectoryIndex]})
 	if err != nil {
 		return nil, err
 	}
+	if inputs[startDistanceAlongTrajectoryIndex].Value != 0 {
```

Comment:
This is kind of weird.  I don't love that one frame is doing double duty depending on what is passed in as inputs

Reply:
I don't understand how we're doing double duty here? This is just a minor optimization.

The return of `Transform` is the `PoseBetween` of `ptg(alpha, startDist)` and `ptg(alpha, endDist)`

In the case where startDist is 0, we can save a bit of computation because that pose is always the zero pose, so we don't need to do the extra `Transform` and then `PoseBetween`. The `if` statement could be removed and we could run the `PoseBetween` on all inputs and there would be no output change.

Reply:
I was misreading this before.  This makes sense now